sample/src/main/java/com/google/accompanist/sample/navigation/animation/AnimatedNavHostSample.kt

```diff
@@ -0,0 +1,295 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.sample.navigation.animation
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.expandIn
+import androidx.compose.animation.shrinkOut
+import androidx.compose.animation.slideInHorizontally
+import androidx.compose.animation.slideInVertically
+import androidx.compose.animation.slideOutHorizontally
+import androidx.compose.animation.slideOutVertically
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.unit.Dp
+import androidx.navigation.NavController
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.plusAssign
+import com.google.accompanist.navigation.animation.AnimatedComposeNavigator
+import com.google.accompanist.navigation.animation.AnimatedNavHost
+import com.google.accompanist.navigation.animation.composable
+import com.google.accompanist.navigation.animation.navigation
+import com.google.accompanist.sample.AccompanistSampleTheme
+
+@ExperimentalAnimationApi
+class AnimatedNavHostSample : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            AccompanistSampleTheme {
+                ExperimentalAnimationNav()
+            }
+        }
+    }
+}
+
+@ExperimentalAnimationApi
+@Composable
+fun ExperimentalAnimationNav() {
+    val navController = rememberNavController()
+    navController.navigatorProvider += AnimatedComposeNavigator()
```

Comment:
This is going to happen on every composition. Shouldn't this be in an effect?

``` kotlin
DisposableEffect(navController) {
    navController.navigatorProvider += AnimatedComposeNavigator()
    onDispose { ... }
}
```

Reply:
Also, could we make `AnimatedNavHost()` do this automatically? Feels like something which devs shouldn't need to think about. We could then make `AnimatedComposeNavigator` internal?

Reply:
No, this needs to happen immediately (and is idempotent and safe to do every composition). The request for making an official API for this pattern is being tracked in [b/193555369](https://issuetracker.google.com/193555369).

We could certainly make a `rememberAnimatedNavController()` API to encapsulate this work, but this work does need to happen outside of the NavHost.

Reply:
Made the `rememberAnimatedNavController()` API and used that instead.